modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/HeaderInformation.java

```diff
@@ -0,0 +1,823 @@
+package de.wuespace.telestion.api.message;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.eventbus.Message;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * <h2>Description</h2>
+ *
+ * <p>
+ * Header information are useful to attach simple information to messages
+ * that should be transferred over the Vert.x event bus.
+ * <p>
+ * This implementation extends the already existing functionality
+ * of the {@link Message Vert.x message} {@link MultiMap headers}
+ * to allow storing and retrieving of other basic data types than {@link String}.
+ * <p>
+ * There are numerous interfaces that simplifies the usage with the existing features of Vert.x
+ * and better support through the {@link de.wuespace.telestion.api.verticle.trait.WithEventBus WithEventBus trait}
+ * for {@link io.vertx.core.Verticle Vert.x verticles}
+ * and especially the {@link de.wuespace.telestion.api.verticle.TelestionVerticle TelestionVerticle}.
+ *
+ * <h2>Usage</h2>
+ * <pre>
+ * {@code
+ * public class MyVerticle extends TelestionVerticle<GenericConfiguration> implements WithEventBus {
+ *     @Override
+ *     public void onStart() {
+ *         var requestInfos = new HeaderInformation()
+ *                 .add("version", 1)
+ *                 .add("log-enabled", true)
+ *                 .add("my-initial-char", 'C').
+ *                 .add("my-coworkers-name", "fussel178");
+ *
+ *         request("request-address", "Hello with Info", requestInfos).onSuccess(message -> {
+ *             var responseInfos = HeaderInformation.from(message);
+ *
+ *             logger.info("Received version: {}", responseInfos.getInt("version", -1));
+ *             logger.info("Received log enabled state: {}", responseInfos.getBoolean("log-enabled", false));
+ *             logger.info("Received initial char: {}", responseInfos.getChar("my-initial-char", '\0'));
+ *             logger.info("Received coworker name: {}", responseInfos.get("my-coworkers-name", "unset"));
+ *         });
+ *     }
+ * }
+ * }
+ * </pre>
+ *
+ * @author Cedric Boes (@cb0s), Ludwig Richter (@fussel178)
+ */
+public class HeaderInformation {
+
+	/**
+	 * Merges an array of header information into one header information object.
+	 *
+	 * @param information multiple header information that should be merged into one
+	 * @return the merged header information
+	 */
+	public static HeaderInformation merge(HeaderInformation... information) {
+		return merge(Arrays.stream(information));
+	}
+
+	/**
+	 * Merges a list of header information into one header information object.
```

Comment:
```suggestion
	 * Merges a list of header information objects into one header information object.
```

Reply:
Done in [`2cd29d6` (#484)](https://github.com/wuespace/telestion-core/pull/484/commits/2cd29d6f1426056b991dcbdcda1b3d2c3274fe9e)